dataland-frontend/tests/e2e/fixtures/frameworks/lksg/LksgGenerator.ts

```diff
@@ -0,0 +1,66 @@
+import { Generator } from "@e2e/utils/FakeFixtureUtils";
+import { type LksgProduct } from "@clients/backend/org/dataland/datalandfrontend/openApiClient/backend/model/lksg-product";
+import { type LksgProcurementCategory, type LksgProductionSite } from "@clients/backend";
+import { ProcurementCategoryType } from "@/api-models/ProcurementCategoryType";
+import { generateAddress } from "@e2e/fixtures/common/AddressFixtures";
+import { faker } from "@faker-js/faker";
+export class LksgGenerator extends Generator {
+  generateLksgProduct(): LksgProduct {
+    return {
+      name: this.guaranteedShortString(),
+      productionSteps: this.randomArray(() => this.guaranteedShortString(), 0, 5),
+      relatedCorporateSupplyChain: this.randomShortString(),
+    };
+  }
+
+  //TODO generalize with a template
+  generateCustomRandomMap(numEntries: number): { [p: string]: number } {
+    const map = new Map<string, number>();
+    const randomStringArray = this.guaranteedArray(() => this.guaranteedShortString(), 0, numEntries);
+    randomStringArray.forEach((stringEntry) => map.set(stringEntry, this.guaranteedInt()));
+    return Object.fromEntries(map);
+  }
+
+  generateLkSGProcurementCategory(localNullProbability: number): LksgProcurementCategory {
```

Comment:
localNullProbability for this and the next function is unused except for the default parameter value. It should be dropped

Reply:
done